sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala

```diff
@@ -40,17 +41,31 @@ object FileStreamSink extends Logging {
    * be read.
    */
   def hasMetadata(path: Seq[String], hadoopConf: Configuration, sqlConf: SQLConf): Boolean = {
-    path match {
-      case Seq(singlePath) =>
-        val hdfsPath = new Path(singlePath)
-        val fs = hdfsPath.getFileSystem(hadoopConf)
-        if (fs.isDirectory(hdfsPath)) {
-          val metadataPath = getMetadataLogPath(fs, hdfsPath, sqlConf)
-          fs.exists(metadataPath)
-        } else {
-          false
-        }
-      case _ => false
+    if (sqlConf.getConf(SQLConf.FILE_SINK_FORMAT_CHECK_ENABLED)) {
+      path match {
+        case Seq(singlePath) =>
+          val hdfsPath = new Path(singlePath)
+          try {
+            val fs = hdfsPath.getFileSystem(hadoopConf)
+            if (fs.isDirectory(hdfsPath)) {
+              val metadataPath = getMetadataLogPath(fs, hdfsPath, sqlConf)
+              fs.exists(metadataPath)
```

Comment:
does this need to be a file/dir, or is existence all that matters?

Reply:
existence is all that matters